### PR TITLE
Hardhat readme

### PR DESCRIPTION
**Pull request by donoso-eth**
Readme updated with hardhat plugin infos

### CODE REVIEW

**Review by goums — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -232,116 +239,63 @@ To run your web3 function using mock storage values, add a `storage.json` in you
...
-  Benchmark result:
-  - nb success: 100/100
-  - duration: 64s
-  ```
+The Web3 Function Hardhat Plugin provides built-in hardhat tasks that wil speed your development as well as provide a great DevX for end to end testing.
 
+In order to user the Hardhat Plugin you will need to:
 
+### Config Hardhat 
+Config `hardhat.config.ts` Web3 Functions Plugin
```
Comment (goums) [inline README.md:251]:
```suggestion
Import Web3 functions Hardhat plugin into `hardhat.config.ts`:  
```

File: README.md [hunk 2]
```diff
@@ -232,116 +239,63 @@ To run your web3 function using mock storage values, add a `storage.json` in you
...
+  networks: ["mumbai", "goerli", "baseGoerli"], //(multiChainProvider) injects provider for these networks
+},
+....
+```
 
-      src/web3-functions/fails/syntax-error.js:1:30:
-        1 │ import { Web3Function } from "nothing";
-          ╵                               ~~~~~~~~~
+### Web3 Function Testing 
+Use following command your your tests:
```
Comment (goums) [inline README.md:266]:
```suggestion
Use the following command to run your test:
```

File: README.md [hunk 3]
```diff
@@ -232,116 +239,63 @@ To run your web3 function using mock storage values, add a `storage.json` in you
...
 
-- Web3Function ends without returning result:
-  - Run: `yarn test src/web3-functions/fails/no-result.ts`
-  - Result:
-  ```
-  Web3Function Result:
-  ✗ Error: Web3Function exited without returning result
-  ```
+```ts
+  const { w3f } = hre;
```
Comment (goums) [inline README.md:286]: Let's reindent properly the example

```suggestion
const { w3f } = hre;
```

**Comment by donoso-eth** [comment]
Changes done as requested

**Review by goums — APPROVED** [review]

**Pull request merged**